src/app/ui/sequence.tsx

```diff
@@ -0,0 +1,860 @@
+    );
+
+    if (duplicates.length > 0) {
+        const unique = Array.from(new Set(duplicates));
+        const duplicationsPerEntityPerDuplicatedChain = unique.map(
+            (duplicatedChainId) => {
+                const coincidencesPerEntity = chainOptions.flatMap((c) => {
+                    const coincidencesChain = c.chains
+                        .filter(
+                            ([_id, label]) =>
+                                label.replace(chainIdRegex, "$1$2") ===
+                                duplicatedChainId
+                        )
+                        .map(([_id, label]) =>
+                            label.replace(chainIdRegex, "$1 [auth $2]")
+                        );
+                    if (coincidencesChain.length > 0)
+                        return {
+                            entityId: c.entityId,
+                            chainsDuplicated: coincidencesChain.join(", "),
+                        };
+                    else return [];
+                });
+
+                return { duplicatedChainId, coincidencesPerEntity };
+            }
+        );
+
+        // Maintenance Note: Should make tests for all PDBs for this
+        throw new Error(
+            duplicationsPerEntityPerDuplicatedChain
+                .map((i) => {
+                    const coincidencesStr = i.coincidencesPerEntity
+                        .map(
+                            (opt) =>
+                                `EntityId: ${opt.entityId}, Chains: ${opt.chainsDuplicated}`
+                        )
+                        .join("; ");
+                    return `Duplicated chains for: ${i.duplicatedChainId} -> ${coincidencesStr}`;
+                })
+                .join("\n")
+        );
+    }
+
+    return { entityOptions, chainOptions };
+}
+
+// const chainIdRegex = /(?:(\w+) ){0,1}(?:\[auth (\w+)\]){0,1}/; if both ids are present this
+// regex will match (which is not intended for later substitutions).
+// With next regex, if some id is not present will not match, but intended for later substitutions;
+// because if there is no id is because they are the same or there is no struct asym id for formats
+// like "pdb" or "ent"
+const chainIdRegex = /^(?:(\w+)\s{0,1}){0,1}(?:\[auth (\w+)\])$/; // $1 structAsymId, $2 chainId (auth)
+
+export function getEntityIdFromChainId(
+    chainOptions: EntityChainPairs["chainOptions"],
+    chainId: string
+): string {
+    const entityId = chainOptions.find((c) =>
+        c.chains.find(
+            ([_id, label]) => label.replace(chainIdRegex, "$2") === chainId
+        )
+    )?.entityId;
+    if (!entityId) throw new Error(`Entity not found for chain ${chainId}`);
+
+    return entityId;
+}
+
+export function getChainNumberedIdFromChainId(
+    chainOptions: EntityChainPairs["chainOptions"],
+    chainId: string
+): number {
+    const chainNumberedId = chainOptions.reduce<number | undefined>(
```

Comment:
This use of reduce looks off (99% of times, we don't need reduce). It should be doable with typical filter/map/first/...